### DISCUSSION TITLE
Artisan + Phidgets + Ununtu Linux ..Not working

### DISCUSSION THREAD

**Discussion by balliet** [artisan-roaster-scope/artisan#811] [Q&A]
I recently purchaed VINT HUB0001_0  and TMP1101_1 and 2 thermocouples. 

I am using Ubuntu 20.04

I installed all of the hardware according to the instrctions and passed the HelloWorld test to make sure everything was working. 

I changed the 99-libphidget22.rules file to the following:
SUBSYSTEMS=="spi", DRIVERS=="phidget_hub0001_0", ACTION=="add", MODE="666"

After entering Artisan and changing the device and machine setting to use the pfidgets, I thought I was ready to go.

clicked the default ON button.  it display showed that "SCOPE HAS BEEN RESET"  BT and ET was displayed  but there was no  temperature reading displayed.  

I have been working on this for several days and have reloaded the system anew, and reinstalled the Phidgets software again and I get the same results

Need Help .... Thank you

**Comment by MAKOMO** [comment]
I can confirm the issue. The current Artisan v2.4.6 does not support the new firmware 111 of the HUB0001 (as well as the latest firmware 307 of the HUB0000 (at least on Linux platforms, incl. the RPi). The cause is an incompatibility of the outdated Python glue code to access the Phidget driver build into Artisan v2.4.6.

I created a corresponding issue (Issue #812) and started working on this.

**Comment by MAKOMO** [comment]
The root cause for those troubles are that the Linux build of Artisan v2.4.6 contained a bundled Phidget driver which now is outdated and not supporting the HUB0001. Installing a newer system-wide Phidget driver did not help as Artisan prefers the bundled one.

A new development build of Artisan without the bundled driver is available now here:

https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

Install this version and check the Artisan about box for the Phidget driver version. It should report "Phidget22 version 1.9", the version of the driver you installed.

Please report if this works for you with this change now. Thanks!

**Reply by balliet** [reply]
Thank You


Version 2.4.7 for linux  <deb>  works perfectly  so far...uninstalled 2.4.6 and the installed 2.4.7.    did not have to do anything to the Phidgets software

So grateful for you response and guidance.


Thank You,,, Thank you  ,,,,So much

**Reply by MAKOMO** [reply]
Thanks for this report. You didn't need to install anything further, as you already installed the latest Phidget driver before and this version of Artisan is now using the system installed Phidget driver.
